docs/configuring.md

```diff
@@ -1146,43 +1146,53 @@ Like [`message_before_copy`][message_after_copy] but printed before
+    is specified the migration will run on every update.
+-   **when** (optional): Specifies a condition that needs to hold for the task to run.
+    By default, a migration will run in the after upgrade stage.
+-   **working_directory** (optional): Specifies the directory in which the command will
+    be run. Defaults to the destination directory.
+
+If a `str` or `List[str]` is given as a migrator it will be treated as `command` with
+all other items not present.
 
 Migrations will run in the same order as declared here (so you could even run a
 migration for a higher version before running a migration for a lower version if the
 higher one is declared before and the update passes through both).
 
-They will only run when _new version >= declared version > old version_. And only when
-updating (not when copying for the 1st time).
+When `version` is given they will only run when _new version >= declared version > old
+version_. Your template will only be marked as [unsafe][unsafe] if this condition is
+true. Migrations will also only run when updating (not when copying for the 1st time).
 
 If the migrations definition contains Jinja code, it will be rendered with the same
 context as the rest of the template.
 
-Migration processes will receive these environment variables:
-
--   `$STAGE`: Either `before` or `after`.
--   `$VERSION_FROM`: [Git commit description][git describe] of the template as it was
-    before updating.
--   `$VERSION_TO`: [Git commit description][git describe] of the template as it will be
-    after updating.
--   `$VERSION_CURRENT`: The `version` detector as you indicated it when describing
-    migration tasks.
--   `$VERSION_PEP440_FROM`, `$VERSION_PEP440_TO`, `$VERSION_PEP440_CURRENT`: Same as the
-    above, but normalized into a standard [PEP 440][] version string indicator. If your
-    scripts use these environment variables to perform migrations, you probably will
-    prefer to use these variables.
+There are a number of additional variables available for templating of migrations. Those
+variables are also passed to the migration process as environment variables. Migration
+processes will receive these variables:
+
+-   `_stage`/`$STAGE`: Either `before` or `after`.
+-   `_version_from`/`$VERSION_FROM`: [Git commit description][git describe] of the
+    template as it was before updating.
+-   `_version_to`/`$VERSION_TO`: [Git commit description][git describe] of the template
+    as it will be after updating.
+-   `_version_current`/`$VERSION_CURRENT`: The `version` detector as you indicated it
+    when describing migration tasks (only when `version` is given).
+-   `_version_pep440_from`/`$VERSION_PEP440_FROM`,
+    `_version_pep440_to`/`$VERSION_PEP440_TO`,
+    `_version_pep440_current`/`$VERSION_PEP440_CURRENT`: Same as the above, but
+    normalized into a standard [PEP 440][] version. In Jinja templates these are
+    represented as `packaging.version.Version` objects and allow access to their
```

Comment:
suggestion: it'd be cool to link to docs of this object if possible.

```suggestion
    represented as [packaging.version.Version](https://packaging.pypa.io/en/stable/version.html#packaging.version.Version) objects and allow access to their
```

Reply:
Done!